pkgs/top-level/haskell-packages.nix

```diff
@@ -71,29 +71,27 @@ in {
     };
 
     ghc884 = callPackage ../development/compilers/ghc/8.8.4.nix {
-      # the oldest ghc with aarch64-darwin support is 8.10.5
-      bootPkgs = if stdenv.isDarwin && stdenv.isAarch64 then
-          packages.ghc8107BinaryMinimal
+      bootPkgs =
         # aarch64 ghc865Binary gets SEGVs due to haskell#15449 or similar
+        # the oldest ghc with aarch64-darwin support is 8.10.5
         # Musl bindists do not exist for ghc 8.6.5, so we use 8.10.* for them
-        else if stdenv.isAarch64 || stdenv.targetPlatform.isMusl then
-          packages.ghc8102BinaryMinimal
+        if stdenv.isAarch64 || stdenv.targetPlatform.isMusl then
```

Comment:
@nh2 I believe `hostPlatform.isMusl` would be more correct?

Reply:
Yes, I think I did that wrong in https://github.com/NixOS/nixpkgs/commit/8b15fccf8ae1dfb0c9404e46b9b008d15f6a372d

I wasn't sure of it back at the time: https://github.com/NixOS/nixpkgs/pull/129289#issue-936602440

> ###### Extra review needed for
>
> * My usage of `buildPlatform`, `targetPlatform` etc.